### PR TITLE
use etna::redirect

### PR DESCRIPTION
**Pull request by graft**
This adds + uses Etna::Redirect, which has some safety-checking to make sure that the route being redirected to is (1) https and (2) comes from this subdomain. This should trap any attempts to inject bad redirects via refer=, etc.

In the event that the redirect is bad, this returns a 422, which was my best guess for response code.

### CODE REVIEW

File: etna/packages/etna-py/build/lib/mountetna/etna.py [hunk 1]
```diff
@@ -0,0 +1,3 @@
+class EtnaClient:
```
Comment (corps) [inline etna/packages/etna-py/build/lib/mountetna/etna.py:1]: is all of this intentional?

File: etna/lib/etna/redirect.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    end
+
+    private
+
+    def matches_domain?(path)
+      top_domain(@request.server_name) == top_domain(URI(path).host)
+    end
+
+    def top_domain(host_name)
+      host_name.split('.')[-2..-1].join('.')
```
Comment (coleshaw) [inline etna/lib/etna/redirect.rb:33]: This does not seem to work on development, which is unfortunate?

```
Puma caught this error: undefined method `join' for nil:NilClass (NoMethodError)
/etna/lib/etna/redirect.rb:33:in `top_domain'
/etna/lib/etna/redirect.rb:29:in `matches_domain?'
/etna/lib/etna/redirect.rb:15:in `to'
/etna/lib/etna/auth.rb:68:in `fail_or_redirect'
/etna/lib/etna/auth.rb:19:in `call'
/etna/lib/etna/symbolize_params.rb:10:in `call'
/etna/lib/etna/parse_body.rb:38:in `call'
/etna/lib/etna/metrics.rb:22:in `call'
/etna/lib/etna/cross_origin.rb:34:in `actual_response'
/etna/lib/etna/cross_origin.rb:11:in `call'
/app/vendor/bundle/ruby/2.5.0/gems/puma-5.0.2/lib/puma/configuration.rb:238:in `call'
/app/vendor/bundle/ruby/2.5.0/gems/puma-5.0.2/lib/puma/server.rb:638:in `block in handle_request'
/app/vendor/bundle/ruby/2.5.0/gems/puma-5.0.2/lib/puma/thread_pool.rb:337:in `with_force_shutdown'
/app/vendor/bundle/ruby/2.5.0/gems/puma-5.0.2/lib/puma/server.rb:637:in `handle_request'
/app/vendor/bundle/ruby/2.5.0/gems/puma-5.0.2/lib/puma/server.rb:394:in `process_client'
/app/vendor/bundle/ruby/2.5.0/gems/puma-5.0.2/lib/puma/server.rb:250:in `block in run'
/app/vendor/bundle/ruby/2.5.0/gems/puma-5.0.2/lib/puma/thread_pool.rb:145:in `block in spawn_thread'
```

Trying to load Vulcan with no cookie set, from https://vulcan.development.local

Comment (graft) [reply etna/lib/etna/redirect.rb:33]: should be fixed?

**Comment by coleshaw** [comment]
Yes, seems to work now! :+1:

**Comment by corps** [comment]
Shall we merge?  I'll get a prod deploy out soon.

**Pull request merged**